labgrid/driver/shelldriver.py

```diff
@@ -258,35 +262,37 @@ def _put_ssh_key(self, keyfile_path):
 
         if test_write == 0 and read_keys == 0:
             self.logger.debug("Key not on target and writeable, concatenating...")
-            self._write_key(keyline, "~/.ssh/authorized_keys")
+            self._write_key(keyline, dest_authorized_keys)
             self._run_check("rm ~/.test")
             return
 
         if test_write == 0:
-            self.logger.debug("Key not on target, testing for .ssh directory")
-            _, _, ssh_dir = self._run("[ -d ~/.ssh/ ]")
+            self.logger.debug("Key not on target, testing for % directory", dest_authorized_keys_dir)
+            _, _, ssh_dir = self._run(f"""[ -d {dest_authorized_keys_dir} ]""")
             if ssh_dir != 0:
-                self.logger.debug("~/.ssh did not exist, creating")
-                self._run("mkdir ~/.ssh/")
-            self._run_check("chmod 700 ~/.ssh/")
-            self.logger.debug("Creating ~/.ssh/authorized_keys")
-            self._run_check("touch ~/.ssh/authorized_keys")
-            self._write_key(keyline, "~/.ssh/authorized_keys")
+                self.logger.debug(" % did not exits, creating", dest_authorized_keys_dir)
+                self._run(f"""mkdir -p {dest_authorized_keys_dir}""")
+            self._run_check(f"""chmod 700 {dest_authorized_keys_dir}""")
+            self.logger.debug("Creating %s", dest_authorized_keys)
+            self._write_key(keyline, dest_authorized_keys)
             self._run_check("rm ~/.test")
             return
 
         self.logger.debug("Key not on target and not writeable, using bind mount...")
         self._run_check('mkdir -p -m 700 /tmp/labgrid-ssh/')
-        self._run("cp -a ~/.ssh/* /tmp/labgrid-ssh/")
+        self._run(f"""cp -a {dest_authorized_keys_dir}/* /tmp/labgrid-ssh/""")
         self._write_key(keyline, "/tmp/labgrid-ssh/authorized_keys")
         self._run_check('chmod 600 /tmp/labgrid-ssh/authorized_keys')
-        out, err, exitcode = self._run('mount --bind /tmp/labgrid-ssh/ ~/.ssh/')
+        out, err, exitcode = self._run(f"""mount --bind /tmp/labgrid-ssh/ {dest_authorized_keys_dir}""")
         if exitcode != 0:
-            self.logger.warning("Could not bind mount ~/.ssh directory: %s %s", out, err)
+            self.logger.warning("Could not bind mount %s directory: %s %s", 
+                                dest_authorized_keys_dir, out, err)
 
     @Driver.check_active
-    def put_ssh_key(self, keyfile_path):
-        self._put_ssh_key(keyfile_path)
+    def put_ssh_key(self, keyfile_path, dest_authorized_keys = None):
```

Comment:
While moving the documentation comment, please extend it for all arguments of the function.